rust/src/python/setup.py

```diff
@@ -1,18 +1,35 @@
 # SPDX-License-Identifier: LGPL-2.1-or-later
-
+import ctypes
+import sys
 import setuptools
 
 
 def requirements():
     req = []
-    with open("requirements.txt") as fd:
+    with open("requirements.txt", encoding="utf-8") as fd:
         for line in fd:
             line.strip()
             if not line.startswith("#"):
                 req.append(line)
     return req
 
 
+def check_nmstate_c_library():
+    if "bdist_rpm" not in sys.argv:
+        try:
+            ctypes.cdll.LoadLibrary("libnmstate.so.2")
+        except OSError as exc:
+            raise RuntimeError(
+                "Error: nmstate C library not found. "
+                "Please install nmstate C library separately "
+                "before installing the Python package."
+                "See: https://github.com/nmstate/nmstate"
```

Comment:
I followed the link  https://github.com/nmstate/nmstate, but I did not see the link contain the instruction on installing the c library.

Reply:
If you look at the Makefile, you will notice that the c library is installed by compiling the project.

Reply:
Please use https://nmstate.io/user/install.html

Reply:
> If you look at the Makefile, you will notice that the c library is installed by compiling the project. 

updated. but ci is not happy yet